File: Source/Android/luascriptcore/src/main/java/cn/vimfung/luascriptcore/LuaTuple.java
package cn.vimfung.luascriptcore;

import java.util.ArrayList;

/**
 * Created by vimfung on 17/1/19.
 * 元组，用于多个返回值的返回时使用
 */
public class LuaTuple extends LuaBaseObject
{
    private ArrayList<LuaValue> _returnValues;

    public LuaTuple ()
    {
        _returnValues = new ArrayList<LuaValue>();
    }

    /**
     * 元素数量
     * @return 数量
     */
    public int count()
    {
        return _returnValues.size();
    }

    /**
     * 添加返回值
     * @param value 返回值
     */
    public void addReturnValue(Object value)
    {
        _returnValues.add(new LuaValue(value));
    }

    /**
     * 获取返回值
     * @param index 返回值索引
     * @return  返回值
     */
    public Object getReturnValueByIndex(int index)
    {
        return _returnValues.get(index).toObject();
    }
}

File: Source/iOS_OSX/Code-Swift/LuaTuple.swift
//
//  LuaTuple.swift
//  LuaScriptCore
//
//  Created by 冯鸿杰 on 17/1/19.
//  Copyright © 2017年 vimfung. All rights reserved.
//

public class LuaTuple: NSObject
{
    var _rawTuple : LSCTuple;
    
    /// 初始化
    public override init ()
    {
        _rawTuple = LSCTuple();
    }
    
    /// 初始化
    ///
    /// - Parameter rawTuple: OC中的Lua元组对象实例
    public init(rawTuple : LSCTuple)
    {
        _rawTuple = rawTuple;
    }
    
    /// 获取元组数量
    public func count() -> Int
    {
        return _rawTuple.count;
    }
    
    /// 添加返回值
    ///
    /// - Parameter value: 返回值
    public func addReturnValue (value : Any)
    {
        _rawTuple.addReturnValue(value);
    }
    
    /// 获取返回值
    ///
    /// - Parameter index: 索引
    /// - Returns: 返回值
    public func returnValueForIndex (index : Int) -> Any
    {
        return _rawTuple.returnValue(for: index);
    }
}
